packit/schema.py

```diff
@@ -483,18 +488,38 @@ class PackageConfigSchema(CommonConfigSchema):
     """
 
     jobs = fields.Nested(JobConfigSchema, many=True)
+    packages = fields.Dict(
+        keys=fields.String(), values=fields.Nested(CommonConfigSchema())
+    )
 
     # list of deprecated keys and their replacement (new,old)
     deprecated_keys = (("upstream_package_name", "upstream_project_name"),)
 
     @pre_load
-    def ordered_preprocess(self, data, **kwargs):
-        data = self.rename_deprecated_keys(data, **kwargs)
+    def ordered_preprocess(self, data: dict, **_) -> dict:
+        # Create a deepcopy(), so that loading doesn't modify the
+        # dictionary received.
+        data = copy.deepcopy(data)
+        data = self.rename_deprecated_keys(data)
+        # Don't use 'setdefault' in this case, as we should expect
+        # downstream_package_name only if there is no 'packages' key.
+        if "packages" not in data:
+            package_name = data.pop("downstream_package_name")
+            paths = data.pop("paths", ["./"])
+            data["packages"] = {
+                package_name: {
+                    "downstream_package_name": package_name,
+                    "paths": paths,
+                }
+            }
```

Comment:
does it make sense to also test that 'paths' is populated correctly?

Reply:
`test_loading_packageless_config` should cover the aspect when `paths` is not set, and `test_multiple_packages_in_jobs` should cover the aspect when `paths` is set. Should I add any other test case?

Reply:
oki, understood; I was really just grepping the "paths" word in tests. Those two test cases you described are sufficient to me.